test/membrane/blank_video_generator_test.exs

```diff
@@ -0,0 +1,100 @@
+defmodule Membrane.BlankVideoGeneratorTest do
+  use ExUnit.Case, async: true
+  use Membrane.Pipeline
+
+  import Membrane.Testing.Assertions
+
+  alias Membrane.{BlankVideoGenerator, Buffer}
+  alias Membrane.Caps.Video.Raw
+  alias Membrane.H264.FFmpeg.Encoder
+  alias Membrane.Testing.{Pipeline, Sink}
+
+  @caps_i420 %Raw{
+    format: :I420,
+    height: 720,
+    width: 1280,
+    framerate: {1, 1},
+    aligned: true
+  }
+  @caps_i422 %Raw{@caps_i420 | format: :I422}
+
+  describe "Blank Video Generator should work with buffers as demand unit for caps" do
+    test "with I420 format" do
+      test_for_caps(@caps_i420)
+    end
+
+    test "with I422 format" do
+      test_for_caps(@caps_i422)
+    end
+  end
+
+  describe "Generated raw video should be successfully encoded to H264 for caps" do
+    test "with I420 format" do
+      test_h264(@caps_i420)
+    end
+
+    test "with I422 format" do
+      test_h264(@caps_i422)
+    end
+  end
+
+  defp test_for_caps(caps) do
+    duration = Membrane.Time.seconds(3)
+
+    elements = [
+      generator: %BlankVideoGenerator{caps: caps, duration: duration},
+      sink: Sink
+    ]
+
+    links = [
+      link(:generator)
+      |> to(:sink)
+    ]
+
+    pipeline_options = %Pipeline.Options{elements: elements, links: links}
+    assert {:ok, pid} = Pipeline.start_link(pipeline_options)
```

Comment:
BTW, breaking the test to generate error message I realized the pipeline is not shutdown properly if the test doesn't reach `stop_and_terminate`, so the way to go would be:
```suggestion
    assert {:ok, pid} = Pipeline.start_link(pipeline_options)
    on_exit(fn -> Pipeline.stop_and_terminate(blocking: true) end)
```